lnbits/static/js/components/extension-settings.js

```diff
@@ -0,0 +1,115 @@
+Vue.component('lnbits-extension-settings-form', {
+  name: 'lnbits-extension-settings-form',
+  props: ['options', 'name', 'cancel'],
+  computed: {
+    endpoint: function () {
+      return `/${this.name}/api/v1/settings?usr=${this.usr}`
+    }
+  },
+  methods: {
+    updateSettings: async function () {
+      try {
+        const {data} = await LNbits.api.request(
+          'PUT',
+          this.endpoint,
+          null,
+          this.settings
+        )
+        this.settings = data
+      } catch (error) {
+        LNbits.utils.notifyApiError(error)
+      }
+    },
+    getSettings: async function () {
+      try {
+        const {data} = await LNbits.api.request('GET', this.endpoint)
+        this.settings = data
+      } catch (error) {
+        LNbits.utils.notifyApiError(error)
+      }
+    },
+    resetSettings: async function () {
+      try {
+        await LNbits.api.request('DELETE', this.endpoint)
+        await this.getSettings()
+      } catch (error) {
+        LNbits.utils.notifyApiError(error)
+      }
+    }
+  },
+  created: async function () {
+    if (window.user.admin) {
+      this.admin = true
+      this.usr = window.user.id
+      this.getSettings()
+    }
+  },
+  template: `
+    <q-form v-if="admin && settings" @submit="updateSettings" class="q-gutter-md">
+      <lnbits-dynamic-fields :options="options" v-model="settings"></lnbits-dynamic-fields>
+      <div class="row q-mt-lg">
+        <q-btn v-close-popup unelevated color="primary" type="submit">Update</q-btn>
+        <q-btn v-close-popup unelevated color="danger" @click="resetSettings" >Reset</q-btn>
+        <q-btn v-if="cancel" v-close-popup flat color="grey" class="q-ml-auto">Cancel</q-btn>
```

Comment:
- what about a ` <slot name="actions"></slot>` with a corresponding `<template v-slot:actions> ...</template>`
- it would make the component more generic, allowing for `Close`, `Export` or other random actions
- it would simplify the component interface (by removing the `cancel` property)